src/Oscoin/Storage/Block/Orphanage.hs

```diff
@@ -105,9 +115,17 @@ fromChainSuffix :: (Block c tx (Sealed c s) -> Score)
                 -> OldestFirst NonEmpty (Block c tx (Sealed c s))
                 -> ChainCandidate c s
 fromChainSuffix scoreBlock (toOldestFirst -> blks) = ChainCandidate
-    { candidateChain = Seq.fromList . NonEmpty.toList . map blockHash $ blks
+    { candidateChain = OldestFirst
+                     . Seq.fromList
+                     . NonEmpty.toList
+                     . map blockHash
+                     $ blks
     , candidateTipHeader = blockHeader . NonEmpty.head $ blks
```

Comment:
I just noticed that this is incorrect. Since we get the oldest block first the tip needs to be the last block. Feel free to fix this here or create an issue.

Reply:
This was very quick to change, so I did amend it as part of this PR. Probably in a separate ticket though we should add a bunch of tests to check these invariants (i.e. that the tip is what advertised and ditto for the rootHash) as it's always a bit disheartening when tests don't catch such bugs, it means we got too few of them 😉